src/dotnet/Chat.Service/ChatsBackend.cs

```diff
@@ -415,10 +414,13 @@ await ownerUserIds
         context.Operation().Items.Set(entry);
         context.Operation().Items.Set(isNew);
 
-        if (!entry.Content.IsNullOrEmpty() && !entry.IsStreaming && entry.Type == ChatEntryType.Text) {
-            var chatEvent = new NewChatEntryEvent(entry.ChatId, entry.Id, entry.AuthorId, entry.Content);
-            await EventPublisher.Publish(chatEvent, cancellationToken).ConfigureAwait(false);
-        }
+        if (!entry.Content.IsNullOrEmpty() && !entry.IsStreaming && entry.Type == ChatEntryType.Text)
+            await new NewTextEntryEvent(entry.ChatId, entry.Id, entry.AuthorId, entry.Content)
+                .Configure()
+                .ShardByChatId(entry.ChatId)
+                .WithPriority(EventPriority.Low)
+                .ScheduleOnCompletion(command)
```

Comment:
I'd use this syntax: `.EnqueueOnCompletion(QueueRef queueRef)`. 

`QueueRef` would typically be constructed as: `Queues.Chats.Shard(chatId).Priority(...)`.

Reply:
As far as I can judge, async API isn't absolutely necessary here assuming we just put this info to e.g. `commandContext.Outermost.Items` & process them either right before or right after the commit.

And if it's going to be async (this is more flexible), I'd expose it as ValueTask for now, + add CancellationToken arg.

Reply:
Why `Queues`? Well, we can statically store all the builders there, which are structs. The builders themselves aren't doing anything but provide extension methods, + once completed, they produce `QueueRef`, which has just `string QueueId` property.

Maybe we don't even need `QueueRef` - i.e. we can agree on `.EnqueueOnCompletion(string queueId, CancellationToken cancellationToken)` syntax too. The key thing here is how they're built.

One other interesting q. is: should this QueueRef be a part of event? IMO not, coz as you've said, technically it might be routed to N queues, which also means `.EnqueueOnCompletion` may have an overload with `string[] queueIds`.

Reply:
Also, on names:
- `ScheduleXxx` usually means "run at certain time / with certain delay / periodically / etc."
- `EnqueueXxx` means "add to queue / run on one of queue processors as soon as possible".

IMO for scheduling (running with a certain delay) we'll need something else later, so I'd avoid this name for now.